drivers/entropy/entropy_renesas_ra.c

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2024 Renesas Electronics Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT renesas_ra_rsip_e51a_trng
+
+#include <soc.h>
+#include <zephyr/drivers/entropy.h>
+
+#include "hw_sce_trng_private.h"
+#include "hw_sce_private.h"
+
+static int entropy_ra_rsip_trng_get_entropy(const struct device *dev, uint8_t *buf, uint16_t len)
+{
+	ARG_UNUSED(dev);
+
+	if (buf == NULL) {
+		return -EINVAL;
+	}
+
+	while (len > 0) {
+		uint32_t n[4];
+		const uint32_t to_copy = MIN(sizeof(n), len);
+
+		HW_SCE_RNG_Read(n);
```

Comment:
```
if (HW_SCE_RNG_Read(n) != FSP_SUCCESS) {
          return -ENODATA;
}
```

If the driver is used to generated CS random data we need to know when the HW couldn't generate random bits.

Reply:
I updated for it

Reply:
Hi @ceolin , Could you please help me check it?

Reply:
@thenguyenyf thanks a lot !